app/models/billing/usage/tracker.rb

```diff
@@ -26,7 +26,12 @@ def track(action, model, count)
   def cycle_as_needed
     return nil unless needs_cycling?
 
-    send(BulletTrain::Billing::Usage.parent_association).billing_usage_trackers.create(duration: duration, interval: interval)
+    send(:trackable).billing_usage_trackers.each(&:cycle)
```

Comment:
Ah I don't think this is right! This will cycle all the trackers (i.e. daily. monthly, yearly) etc, but we only need to cycle the tracker in question. So I think all we need to do is

```suggestion
    cycle
```

(I only picked this up after seeing you loom 😬 )